### PR TITLE
New layers panel

### PR DESCRIPTION
**Pull request by wnordmann**
Adding in new layer panel

### CODE REVIEW

**Comment by bartvde** [comment]
@wnordmann I'll review once you say it's ready for review, TIA

File: js/components/LayerListItem.js [hunk 1]
```diff
@@ -347,23 +347,26 @@ class LayerListItem extends React.PureComponent {
...
-      extension: 'kml'
-    },
-    GPX: {
-      format: new ol.format.GPX(),
-      mimeType: 'application/gpx+xml',
-      extension: 'gpx'
+  getFormats(format) {
+    var formats = {
+      GeoJSON: {
+        format: new ol.format.GeoJSON(),
```
Comment (bartvde) [inline js/components/LayerListItem.js]: won't this recreate all the ol formats multiple times? That's why I had them static.

File: js/components/LayerListItem.js [hunk 2]
```diff
@@ -536,23 +568,32 @@ class LayerListItem extends React.PureComponent {
...
+    }
     var zoomTo;
     if (layer.get('type') !== 'base' && layer.get('type') !== 'base-group' && ((source && source.getExtent) || layer.get('EX_GeographicBoundingBox')) && this.props.showZoomTo) {
-      zoomTo = (<Button className='layer-list-item-zoom' onTouchTap={this._zoomTo.bind(this)} tooltipPosition='top' style={iconStyle} buttonType='Icon' tooltip={formatMessage(messages.zoombuttonlabel)}><ZoomInIcon /></Button>);
+      zoomTo = <i className="fa fa-crosshairs" onTouchTap={this._zoomTo.bind(this)}></i>;
     }
     var download;
     if (layer instanceof ol.layer.Vector && this.props.showDownload) {
-      download = (<Button className='layer-list-item-download' onTouchTap={this._download.bind(this)} tooltipPosition='top' style={iconStyle} buttonType='Icon' tooltip={formatMessage(messages.downloadbuttonlabel)}><DownloadIcon /></Button>);
+      download = <MenuItem primaryText='Download' leftIcon={<Download />}onTouchTap={this._download.bind(this)} />
```
Comment (bartvde) [inline js/components/LayerListItem.js]: we should i18n primaryText

File: js/components/LayerListItem.js [hunk 3]
```diff
@@ -611,25 +640,62 @@ class LayerListItem extends React.PureComponent {
...
+      checked={this.state.checked}
+      value={this.props.title}
+      onClick={this._handleVisibility.bind(this)}
+      disableTouchRipple={true}/>);
+
+    var visibility = this.state.checked ? checked : unchecked;
+
+    var popoverEllipsis = (
+      <div>
+        <i className="fa fa-ellipsis-v" onTouchTap={this._handleMenuOpen}></i>
```
Comment (bartvde) [inline js/components/LayerListItem.js]: doesn't this need a ```.bind(this)```  ?

File: js/components/LayerListItem.js [hunk 4]
```diff
@@ -611,25 +640,62 @@ class LayerListItem extends React.PureComponent {
...
+        <Popover
+            open={this.state.popover}
+            anchorEl={this.state.anchorEl}
+            anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}
+            targetOrigin={{horizontal: 'right', vertical: 'top'}}
+            onRequestClose={this._handleRequestClose}
```
Comment (bartvde) [inline js/components/LayerListItem.js]: same wrt bind

File: js/components/LayerListItem.js [hunk 5]
```diff
@@ -561,28 +602,16 @@ class LayerListItem extends React.PureComponent {
...
     }
     var edit;
     if (this.props.allowEditing && layer.get('isWFST') === true) {
       edit = (<Button className='layer-list-item-edit' onTouchTap={this._edit.bind(this)} tooltipPosition='top' style={iconStyle} buttonType='Icon' tooltip={formatMessage(messages.editbuttonlabel)} ><EditIcon /></Button>);
     }
-    var input, labelStyle;
-    if (layer.get('type') === 'base' && this.props.group && this.props.group.get('type') === 'base-group') {
-      input = (<RadioButton disabled={this.state.disabled} checked={this.state.checked} label={this.props.title} value={this.props.title} onCheck={this._handleChange.bind(this)} disableTouchRipple={true}/>);
-    } else {
-      if (this.props.handleResolutionChange && this.props.layer.get('type') !== 'base-group' && !this.calculateInRange()) {
```
Comment (bartvde) [inline js/components/LayerListItem.js]: so does this mean we loose the ability to see when a layer is in scale, or is this now handled differently?

File: js/components/LayerListItem.js (hunk 2 above)
Comment (wnordmann) [inline js/components/LayerListItem.js]: Fixed, on filter other primarytext

File: js/components/LayerListItem.js (hunk 3 above)
Comment (wnordmann) [inline js/components/LayerListItem.js]: correct

File: js/components/LayerListItem.js (hunk 4 above)
Comment (wnordmann) [inline js/components/LayerListItem.js]: fixed

**Comment by bartvde** [comment]
Ah you're right, we must have lost labelStyle somewhere earlier before https://github.com/boundlessgeo/sdk/commit/8f7ab5ff84d4900964f704e143e2dc2ccf0e2c49 ok let's leave it out for now, we can always bring it back later

File: package.json [hunk 6]
```diff
@@ -10,15 +10,17 @@
   },
   "scripts": {
     "build:babel": "NODE_ENV=release babel ./js --out-dir ./release",
-    "start": "parallelshell 'npm run build:copy:css' 'npm run babel' 'npm run watch:css'",
```
Comment (bartvde) [inline package.json]: is this change intentional? I added it a while back with: https://github.com/boundlessgeo/sdk/commit/73ae12d3e521dd1a29b94a29f4c6941793104481

Maybe a mistake in the merge?
